File: Java/Matrizes/matriz_geral.java
// Exercício: Matriz Geral
// Objetivo: Processar uma matriz quadrada realizando soma dos positivos, leitura/exibição de linha e coluna específicas, exibição da diagonal principal e alteração dos valores negativos para seus respectivos quadrados.

import java.util.Locale;
import java.util.Scanner;

public class matriz_geral {
    public static void main(String[] args) {
        
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        int n, linha, coluna;
        double soma;

        System.out.print("Qual a ordem da matriz? ");
        n = sc.nextInt();

        double[][] mat = new double[n][n];

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                System.out.print("Elemento [" + i + "," + j + "]: ");
                mat[i][j] = sc.nextDouble();
            }
        }

        soma = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (mat[i][j] > 0) {
                    soma = soma + mat[i][j];
                }
            }
        }
        System.out.println();

        System.out.print("SOMA DOS POSITIVOS: " + String.format("%.1f", soma));
        System.out.println();

        System.out.println();
        System.out.print("Escolha uma linha: ");
        linha = sc.nextInt();

        System.out.print("LINHA ESCOLHIDA: ");
        for (int j = 0; j < n; j++) {
            System.out.print(String.format("%.1f", mat[linha][j]) + " ");
        }
        System.out.println();

        System.out.println();
        System.out.print("Escolha uma coluna: ");
        coluna = sc.nextInt();

        System.out.print("COLUNA ESCOLHIDA: ");
        for (int i = 0; i < n; i++) {
            System.out.print(String.format("%.1f", mat[i][coluna]) + " ");
        }
        System.out.println();

        System.out.println();
        System.out.print("DIAGONAL PRINCIPAL: ");
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (i == j) {
                    System.out.print(String.format("%.1f", mat[i][j]) + " ");
                }
            }
        }
        System.out.println();

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (mat[i][j] < 0) {
                    mat[i][j] = mat[i][j] * mat[i][j];
                }
            }
        }

        System.out.println();
        System.out.println("MATRIZ ALTERADA:");
        for (int i = 0; i < n; i++) {
            for(int j = 0; j < n; j++) {
                System.out.print(String.format("%.1f", mat[i][j]) + " ");
            }
            System.out.println();
        }

        sc.close();
    }
}

File: Python/matrizes/matriz_geral.py
# Exercício: Matriz Geral
# Objetivo: Ler uma matriz quadrada, calcular soma dos positivos, exibir uma linha e coluna específicas, exibir a diagonal principal e alterar a matriz elevando os negativos ao quadrado.

n = int(input("Qual a ordem da matriz? "))

mat = []

for i in range(n):
    linhas = []

    for j in range(n):
        colunas = float(input(f"Elemento [{i},{j}]: "))
        linhas.append(colunas)
    mat.append(linhas)

soma = 0.0
for i in range(n):
    for j in range(n):
        if mat[i][j] > 0:
            soma = soma + mat[i][j]

print()
print(f"SOMA DOS POSITIVOS: {soma:.1f}")

print()
linha = int(input("Escolha uma linha: "))
print("LINHA ESCOLHIDA:",end=" ")

for j in range(n):
    print(f"{mat[linha][j]:.1f}",end=" ")

print()
print()

coluna = int(input("Escolha uma coluna: "))
print("COLUNA ESCOLHIDA:",end=" ")

for i in range(n):
    print(f"{mat[i][coluna]:.1f}",end= " ")

print()
print()

print("DIAGONAL PRINCIPAL:",end= " ")
for i in range(n):
    for j in range(n):
        if i == j:
            print(f"{mat[i][j]:.1f}",end =" ")

print()
print()

for i in range(n):
    for j in range(n):
        if mat[i][j] < 0:
            mat[i][j] = mat[i][j] * mat[i][j]

print("MATRIZ ALTERADA: ")
for i in range(n):
    for j in range(n):
        print(f"{mat[i][j]:.1f}",end=" ")
    print()            